### PR TITLE
librados: cleanups

### PR DESCRIPTION
**Pull request by tchaikov**
changes from the wishlist in https://pad.ceph.com/p/librados3

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -589,7 +583,7 @@ Description: RADOS striping interface (development files)
 Package: librbd1
 Architecture: linux-any
 Section: libs
-Depends: librados3 (= ${binary:Version}),
```
Comment (jdurgin) [inline debian/control]: @dillaman any issues with CephContext or other structures here?

Comment (dillaman) [reply debian/control]: Yeah -- they need to be version linked

**Comment by jdurgin** [comment]
looks like the make check failure is real:
```../../../lib/librbd_internal.a(internal.cc.o): In function `librbd::tmap_set(librados::IoCtx&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/librbd/internal.cc:285: undefined reference to `librados::IoCtx::tmap_update(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ceph::buffer::list&)'
../../../lib/librbd_internal.a(internal.cc.o): In function `librbd::tmap_rm(librados::IoCtx&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/librbd/internal.cc:294: undefined reference to `librados::IoCtx::tmap_update(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, ceph::buffer::list&)'
../../../lib/librbd_internal.a(RenameRequest.cc.o): In function `librbd::operation::RenameRequest<librbd::ImageCtx>::send_update_directory()':
/home/jenkins-build/build/workspace/ceph-pull-requests/src/librbd/operation/RenameRequest.cc:175: undefined reference to `librados::ObjectWriteOperation::tmap_update(ceph::buffer::list const&)'```

File: debian/control (hunk 1 above)
Comment (tchaikov) [reply debian/control]: ahh, i see. we take advantage of the fact that `librados::config_t` is actually `CephContext*` in `(anonymous_namespace)::get_cct()`.

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-11-04_15:20:53-rados-wip-kefu-testing-2018-11-03-2239-distro-basic-smithi/

**Comment by tchaikov** [comment]
retest this please.

**Comment by tchaikov** [comment]
changelog:

- add a commit to fix the dependency of libradospp-dev: 70a26ed

it should address the failure of http://pulpito.ceph.com/kchai-2018-11-07_07:51:37-rados-wip-kefu2-testing-2018-11-04-1018-distro-basic-smithi/3233734/ , where we have
```
2018-11-07T08:20:28.949 INFO:tasks.workunit.client.0.smithi158.stdout:The following information may help to resolve the situation:
2018-11-07T08:20:28.949 INFO:tasks.workunit.client.0.smithi158.stdout:
2018-11-07T08:20:28.949 INFO:tasks.workunit.client.0.smithi158.stdout:The following packages have unmet dependencies:
2018-11-07T08:20:28.995 INFO:tasks.workunit.client.0.smithi158.stdout: libradospp-dev : Depends: libradospp (= 14.0.0-4787-g9116858-1bionic) but it is not installable
2018-11-07T08:20:29.011 INFO:tasks.workunit.client.0.smithi158.stderr:E: Unable to correct problems, you have held broken packages.
2018-11-07T08:20:29.013 DEBUG:teuthology.orchestra.run:got remote process result: 100
```

**Pull request merged**